lib/wanda_web/views/execution_view.ex

```diff
@@ -0,0 +1,48 @@
+defmodule WandaWeb.ExecutionView do
+  use WandaWeb, :view
+
+  alias Wanda.Results.ExecutionResult
+  alias WandaWeb.ExecutionView
+
+  def render("index.json", %{executions: executions, total_count: total_count}) do
+    %{
+      items: render_many(executions, ExecutionView, "execution.json"),
+      total_count: total_count
+    }
+  end
+
+  def render("show.json", %{execution: execution}) do
+    render_one(execution, ExecutionView, "execution.json")
+  end
+
+  def render("execution.json", %{
+        execution: %ExecutionResult{
+          execution_id: execution_id,
+          group_id: group_id,
+          status: status,
+          payload: result,
+          started_at: started_at,
+          completed_at: completed_at
+        }
+      }) do
+    %{
+      check_results: extract_checks_results(status, result),
+      status: status,
+      started_at: started_at,
+      completed_at: completed_at,
+      execution_id: execution_id,
+      group_id: group_id,
+      result: extract_result(status, result),
+      timeout: extract_timeout(status, result)
+    }
+  end
+
+  defp extract_result(:running, _), do: :unknown
```

Comment:
By the way, I wouldn't use the `unknown` term, because it is not an unknown value. We don't have the result because we are still running, so I would put this value as nil maybe

Reply:
And in other term, in many other places we have named these functions as `map_*` because at the end we are mapping to a new output

Reply:
@arbulu89 @nelsonkopliku +1 for having null and not unknown

Reply:
- made fields nullable
- refactored functions to `map_*`

@fabriziosestito ok with revisiting as you're suggesting, let's keep it in the radar.